requestmanager/requestmanager_test.go

```diff
@@ -831,34 +773,37 @@ func TestPauseResume(t *testing.T) {
 	// verify no further responses come through
 	time.Sleep(100 * time.Millisecond)
 	testutil.AssertChannelEmpty(t, returnedResponseChan, "no response should be sent request is paused")
-	td.fal.CleanupRequest(peers[0], rr.gsr.ID())
 
 	// unpause
 	err = td.requestManager.UnpauseRequest(rr.gsr.ID(), td.extension1, td.extension2)
 	require.NoError(t, err)
 
-	// verify the correct new request with Do-no-send-cids & other extensions
-	resumedRequest := readNNetworkRequests(requestCtx, t, td, 1)[0]
-	doNotSendFirstBlocksData, has := resumedRequest.gsr.Extension(graphsync.ExtensionsDoNotSendFirstBlocks)
-	doNotSendFirstBlocks, err := donotsendfirstblocks.DecodeDoNotSendFirstBlocks(doNotSendFirstBlocksData)
-	require.NoError(t, err)
-	require.Equal(t, pauseAt, int(doNotSendFirstBlocks))
-	require.True(t, has)
-	ext1Data, has := resumedRequest.gsr.Extension(td.extensionName1)
-	require.True(t, has)
-	require.Equal(t, td.extensionData1, ext1Data)
-	ext2Data, has := resumedRequest.gsr.Extension(td.extensionName2)
-	require.True(t, has)
-	require.Equal(t, td.extensionData2, ext2Data)
+	/*
+		TODO: these are no longer used as the old responses are consumed upon restart, to minimize
+		network utilization -- does this make sense? Maybe we should throw out these responses while paused?
```

Comment:
oh, this is interesting .. and I have no idea what the answer to this might be; are you comfortable enough with the new behaviour to commit to it for now?